### PR TITLE
xplat: make sure linker doesn't pass gluing sub libs

### PR DESCRIPTION
**Pull request by obastemur**
- remove unused definitions and fix mixing empty object file (context)
- make sure sub libs (static) are combined (use an empty object file)

### CODE REVIEW

File: bin/ch/WScriptJsrt.cpp [hunk 1]
```diff
@@ -789,8 +789,10 @@ bool WScriptJsrt::Initialize()
     JsValueRef buildValue;
 #ifdef _DEBUG
 #define BUILD_TYPE_STRING_CH "Debug" // (O0)
+#elif !defined(_DEBUG) && defined(ENABLE_DEBUG_CONFIG_OPTIONS)
```
Comment (jianchun) [inline bin/ch/WScriptJsrt.cpp]: Isn't "!defined(_DEBUG)" redundant?

Comment (obastemur) [inline bin/ch/WScriptJsrt.cpp]: Yes, indeed it is. Thanks!

**Comment by jianchun** [comment]
:shipit:

**Comment by obastemur** [comment]
@jianchun Thanks for the review

**Pull request merged**